### PR TITLE
useDeleteThread

### PR DESCRIPTION
**Pull request by FlowFlorent**
### Description

Adds a `useDeleteThread` hook. Only the thread owner can delete the thread. The thread owner is the user of the first comment in the thread.

```tsx
function Example() {
  const { threads } = useThreads();
  const self = useSelf();

  const deleteThread = useDeleteThread();

  return (
    <main>
      {threads.map((thread) => (
        <>
          {self.id === thread.comments[0].userId ? (
            <button onClick={() => deleteThread(thread.id)}>
              Delete thread below
            </button>
          ) : null}
          <Thread key={thread.id} thread={thread} className="thread" />
        </>
      ))}
      <Composer className="composer" />
    </main>
  );
}
```

Screenshot 2024-06-19 at 3 46 42 PM
Screenshot 2024-06-19 at 3 46 30 PM

### CODE REVIEW

**Review by GuillaumeSalles — APPROVED** [review]
Looking good @FlowFlorent! Just missing the docs but the rest looks clean!

File: packages/liveblocks-react/src/room.tsx [hunk 1]
```diff
@@ -538,6 +539,7 @@ function makeRoomContextBundle<
     // context consistent internally.
     return (
       <LiveblocksProviderWithClient client={client} allowNesting>
+        {/* Why type error? */}
```
Comment (nvie) [inline packages/liveblocks-react/src/room.tsx:542]: @FlowFlorent What's this? There should not be any type error, so lmk if this is causing issues for you.

**Review by nvie — APPROVED** [review]
💯

**Review by ofoucherot — APPROVED** [review]

File: packages/liveblocks-react/src/room.tsx (hunk 1 above)
Comment (FlowFlorent) [inline packages/liveblocks-react/src/room.tsx:542]:
```suggestion
```

Comment (FlowFlorent) [reply packages/liveblocks-react/src/room.tsx]: FYI: I removed the comment to tidy up this PR

**Pull request merged**